src/synthesizer/abundances.py

```diff
@@ -245,118 +326,137 @@ class Elements:
 
 class Abundances(Elements):
     def __init__(
-        self, Z=Elements.Z_sol, alpha=0.0, C="Dopita2006", N="Dopita2006", d2m=False
+        self,
+        metallicity=Elements.solar_metallicity,
+        alpha=0.0,
+        abundances=False,
+        carbon_abundance=False,
+        nitrogen_abundance=False,
+        dust_to_metal_ratio=False
     ):
         """
         A class for calcualting elemental abundances including various scaling and depletion on to dust
 
-        Arguments
-        ----------
-        Z : float
-            Mass fraction in metals, default is Solar metallicity.
-        alpha: float
-            Enhancement of the alpha elements
-        C: float, str
-            log10(C/O) ratio. A str is the name of the function f(Z) to use instead.
-        N: float, str
-            log10(N/O) ratio. A str is the name of the function f(Z) to use instead.
-        d2m: float
-            the fraction of metals in dust
+        Arguments:
+            metallicity (float)
+                ass fraction in metals, default is Solar metallicity.
+            alpha (float)
+                Enhancement of the alpha elements.
+            abundances (dict)
+                A dictionary containing the abundances for specific elements or 
+                functions to calculate them for the specified metallicity.
+            dust_to_metal_ratio (float)
+                the fraction of metals in dust.
+
 
-        Returns
-        -------
         """
 
         # save all parameters to object
-        self.Z = Z  # mass fraction in metals
+        self.metallicity = metallicity  # mass fraction in metals
         self.alpha = alpha
-        self.C = C
-        self.N = N
-        self.d2m = d2m
+        self.carbon_abundance = carbon_abundance
+        self.nitrogen_abundance = nitrogen_abundance
+        self.dust_to_metal_ratio = dust_to_metal_ratio
 
         # set depletions to be zero
         self.depletion = {element: 0.0 for element in self.all_elements}
 
-        # Set helium abundance following Bressan et al. (2012)
-        self.Y = 0.2485 + 1.7756 * self.Z
+        # Set helium mass fraction following Bressan et al. (2012)
```

Comment:
Could we add a DOI for this paper? and if possible a description of the values (are these just linear fits?)

Reply:
Added